src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java

```diff
@@ -158,24 +162,55 @@ public void visitToken(DetailAST ast) {
      * @param closingBrace block closing brace
      * @return number of lines with code for current block
      */
-    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
-    @SuppressWarnings("deprecation")
     private int getLengthOfBlock(DetailAST openingBrace, DetailAST closingBrace) {
-        int length = closingBrace.getLineNo() - openingBrace.getLineNo() + 1;
+        final int startLineNo = openingBrace.getLineNo();
+        final int endLineNo = closingBrace.getLineNo();
+        int rawLength = endLineNo - startLineNo + 1;
 
         if (!countEmpty) {
-            final FileContents contents = getFileContents();
-            final int lastLine = closingBrace.getLineNo();
-            // lastLine - 1 is actual last line index. Last line is line with curly brace,
-            // which is always not empty. So, we make it lastLine - 2 to cover last line that
-            // actually may be empty.
-            for (int i = openingBrace.getLineNo() - 1; i <= lastLine - 2; i++) {
-                if (contents.lineIsBlank(i) || contents.lineIsComment(i)) {
-                    length--;
-                }
+            rawLength = countUsedLines(openingBrace.getFirstChild(), startLineNo, rawLength);
```

Comment:
it is very weird set of parameters.

lets make code like:
```
if (countEmpty) {
  length = countUsedLines(openingBrace, closingBrace);
} esle {
   length = getLengthOfBlock(openingBrace, closingBrace);
}
```

Reply:
This way wont work because of pitest.
To calculate length for boolean array, we need to calculate `endLine - startLine + 1`, but pitest will mutate this into `endLine + startLine + 1`, `endLine + startLine - 1` and both will survive because it allocates more space that we need. 
Therefore rawLength is passed as parameter, mutations here will be killed.

To be honest, in this place pitest is only making things uglier.

Reply:
try in this direction:
```
./.ci/pitest.sh pitest-sizes
...
>> Line Coverage: 280/280 (100%)
>> Generated 244 mutations Killed 244 (100%)
>> Mutations with no coverage 0. Test strength 100%
>> Ran 411 tests (1.68 tests per mutation)
Enhanced functionality available at https://www.arcmutate.com/
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  50.196 s
[INFO] Finished at: 2022-02-20T19:39:06-08:00
[INFO] ------------------------------------------------------------------------
✔ ~/java/github/romani/checkstyle [strkkk/11166_remove_getFileContents_method_length L|✚ 1] 
19:39 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java
index 32ccc87..ad9928e 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java
@@ -20,8 +20,12 @@
 package com.puppycrawl.tools.checkstyle.checks.sizes;
 
 import java.util.ArrayDeque;
+import java.util.ArrayList;
 import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
@@ -147,7 +151,14 @@ public class MethodLengthCheck extends AbstractCheck {
         if (openingBrace != null) {
             final DetailAST closingBrace =
                 openingBrace.findFirstToken(TokenTypes.RCURLY);
-            final int length = getLengthOfBlock(openingBrace, closingBrace);
+            final int length;
+            if (countEmpty) {
+                length = getLengthOfBlock(openingBrace, closingBrace);
+            } else {
+                final int startLineNo = openingBrace.getLineNo();
+                final int endLineNo = closingBrace.getLineNo();
+                length = countUsedLines(openingBrace.getFirstChild(), startLineNo);
+            }
             if (length > max) {
                 final String methodName = ast.findFirstToken(TokenTypes.IDENT).getText();
                 log(ast, MSG_KEY, length, max, methodName);
@@ -165,12 +176,7 @@ public class MethodLengthCheck extends AbstractCheck {
     private int getLengthOfBlock(DetailAST openingBrace, DetailAST closingBrace) {
         final int startLineNo = openingBrace.getLineNo();
         final int endLineNo = closingBrace.getLineNo();
-        int rawLength = endLineNo - startLineNo + 1;
-
-        if (!countEmpty) {
-            rawLength = countUsedLines(openingBrace.getFirstChild(), startLineNo, rawLength);
-        }
-        return rawLength;
+        return endLineNo - startLineNo + 1;
     }
 
     /**
@@ -181,9 +187,9 @@ public class MethodLengthCheck extends AbstractCheck {
      * @param length raw length of code block
      * @return number of used lines of code
      */
-    private static int countUsedLines(DetailAST ast, int startLine, int length) {
-        final boolean[] usedLines = new boolean[length];
-        usedLines[0] = true;
+    private static int countUsedLines(DetailAST ast, int startLine) {
+        final Set<Integer> usedLines = new HashSet<Integer>();
+        usedLines.add(0);
         final Deque<DetailAST> nodes = new ArrayDeque<>();
         Stream.iterate(ast, Objects::nonNull, DetailAST::getNextSibling).forEach(nodes::addLast);
         while (!nodes.isEmpty()) {
@@ -196,21 +202,15 @@ public class MethodLengthCheck extends AbstractCheck {
                 // count only lines inside, exclude first and last lines
                 IntStream.range(1, contentLinesCount)
                     .map(idx -> offset + idx)
-                    .forEach(idx -> usedLines[idx] = true);
+                    .forEach(usedLines::add);
             }
             else {
-                usedLines[offset] = true;
+                usedLines.add(offset);
             }
             Stream.iterate(node.getLastChild(), Objects::nonNull, DetailAST::getPreviousSibling)
                 .forEach(nodes::addFirst);
         }
-        int result = 0;
-        for (boolean usedLine : usedLines) {
-            if (usedLine) {
-                result++;
-            }
-        }
-        return result;
+        return usedLines.size();
     }
 
     /**
```

Reply:
> `if (countEmpty)`

should be visitToken method

Reply:
I know that it will work with set, because I used it on some of previous versions.

It is not best approach because we know size of this set in advance, so we can use array of flags and avoid useless overhead - boxing, set extension, creation of useless objects. 

In general, pitest makes implementation very inefficient.

Reply:
Not always, but I agree.
It is also true to jacoco and all other static code analysis